server/src/main/java/org/candlepin/model/OwnerContentCurator.java

```diff
@@ -352,72 +401,72 @@ public void updateOwnerContentReferences(Owner owner, Map<String, String> conten
         // touch anything that even looks like a join. As such, we have to do this in vanilla SQL.
 
-        Session session = this.currentSession();
-
-        // Owner content
-        String sql = "DELETE FROM cp2_owner_content WHERE content_uuid = ?1 AND owner_id = ?2";
-
-        int count = session.createSQLQuery(sql)
-            .setParameter("1", content.getUuid())
-            .setParameter("2", owner.getId())
-            .executeUpdate();
-
-        log.debug("{} owner-content relations updated", count);
-
-        // environment content
-        List<String> ids = session.createSQLQuery("SELECT id FROM cp_environment WHERE owner_id = ?1")
-            .setParameter("1", owner.getId())
-            .list();
-
-        if (ids != null && !ids.isEmpty()) {
-            int inBlocks = ids.size() / AbstractHibernateCurator.IN_OPERATOR_BLOCK_SIZE + 1;
-
-            StringBuilder builder = new StringBuilder("DELETE FROM cp2_environment_content ")
-                .append("WHERE content_uuid = ?1 AND (");
-
-            for (int i = 0; i < inBlocks; ++i) {
-                if (i != 0) {
-                    builder.append(" OR ");
-                }
-
-                builder.append("environment_id IN (?").append(i + 2).append(')');
+        if (contentUuids != null && !contentUuids.isEmpty()) {
+            log.info("Removing owner-content references for owner: {}, {}", owner, contentUuids);
+
+            Session session = this.currentSession();
+
+            // Owner content
+            Map<String, Object> criteria = new HashMap<String, Object>();
+            criteria.put("owner_id", owner.getId());
+            criteria.put("content_uuid", contentUuids);
+
+            int count = this.bulkSQLDelete(OwnerContent.DB_TABLE, criteria);
+            log.info("{} owner-content relations removed", count);
+
+            // product content
+            // Impl note:
+            // We're not managing product-content references, since versioning changes require us to
+            // handle that with more explicit logic. Instead, we rely on the content manager using
+            // the product manager to fork/update products when a related content entity changes.
+
+            // String sql = "SELECT product_uuid FROM " + OwnerProducts.DB_TABLE + " WHERE owner_id = ?1";
+            // List<String> ids = session.createSQLQuery(sql)
+            //     .setParameter("1", owner.getId())
+            //     .list();
+
+            // if (ids != null && !ids.isEmpty()) {
+            //     criteria.clear();
+            //     criteria.put("product_uuid", ids);
+            //     criteria.put("content_uuid", contentUuids);
+
+            //     count = this.bulkSQLDelete(ProductContent.DB_TABLE, criteria);
+            //     log.info("{} product-content relations removed", count);
+            // }
+            // else {
+            //     log.info("0 product-content relations updated");
+            // }
```

Comment:
Is this commented out code left in for any particular reason? Remove it if not necessary.